document building SpinalHDL with mill (#160)
SpinalHDL/SpinalDoc-RTD: source/SpinalHDL/Developers area/mill support.rst

Build through Mill
==================

> SpinalHDL itself can be built by Mill build tools, it could compile/test/publishLocal the existing modules.
> Build through mill can be much faster than sbt, which is useful while debugging.

Reviewer: ```suggestion
SpinalHDL itself can be built with Mill. It can compile/test/publishLocal the existing modules.
Build through mill can be much faster than Sbt, which is useful while debugging.
```

Complie the library
-------------------

.. code-block:: sh

   mill __.compile

>    sbt compile # equivalent alternatives

Reviewer: So it doesn't compile tests?

Author: no, that's why its name is "compile the library"

> Run all test suite
> ------------------

Reviewer: ```suggestion
Run all test suites
-------------------
```

.. code-block:: sh

   mill __.test
   sbt test # equivalent alternatives

> Test a specified suite
> ----------------------

Reviewer: ```suggestion
Run a specified test suite
--------------------------
```

.. code-block:: sh

   mill tester.test.testOnly spinal.xxxxx.xxxxx
   sbt "tester/testOnly spinal.xxxxx.xxxxx" # equivalent alternatives

Run a specified App
-------------------

.. code-block:: sh

   mill tester.runMain spinal.xxxxx.xxxxx
   sbt "tester/runMain spinal.xxxxx.xxxxx" # equivalent alternatives

Publish locally
---------------

> it can also publish the library to local ivy2 repos as a ``dev`` version.

Reviewer: ```suggestion
Mill can also publish the library to the local ivy2 repository as a ``dev`` version.
```

Author: I just notice that github can merge this suggestion easily.
thanks.
